compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -208,25 +209,25 @@ impl<'tcx> ConstToPat<'tcx> {
                 let field = FieldIdx::new(idx);
                 // Patterns can only use monomorphic types.
                 let ty = self.tcx.normalize_erasing_regions(self.typing_env, ty);
-                FieldPat { field, pattern: self.valtree_to_pat(val, ty) }
+                let cv = ty::Value { valtree: val, ty };
```

Comment:
Kinda sad we still end up with the `val, ty` tuple, but also not sure how to do better.

Reply:
Currently the problem is this. In this case it is not an improvement but just refactoring.